src/core/gui/inputdevices/DeviceId.h

```diff
@@ -0,0 +1,22 @@
+/*
+ * Xournal++
+ *
+ * Device identifier
+ *
+ * @author Xournal++ Team
+ * https://github.com/xournalpp/xournalpp
+ *
+ * @license GNU GPLv2 or later
+ */
+
+#pragma once
+
+struct DeviceId {
+    DeviceId() = default;
+    explicit DeviceId(const void* id): id(id) {}
+    void clear() { id = nullptr; }
```

Comment:
Dunno, if we might need this anytime:
```suggestion
    void reset(void const* id = nullptr) { this->id = id; }
```